include/openssl/ssl.h

```diff
@@ -449,7 +451,9 @@ OPENSSL_EXPORT int SSL_write(SSL *ssl, const void *buf, int num);
 // SSL_write_ex writes up to |num| bytes from |buf| into |ssl|. It is similar to
 // |SSL_write|, but instead of returning the number of bytes written, it returns
 // 1 on success or 0 for failure. The number bytes actually written is stored in
-// |written|.
+// |written|. SSL_write_ex() can be called with num=0, but will not send
```

Comment:
Should we add similar behavior to SSL_write() while we are doing this? It's not needed but might be good to add in case another application tries to use this method to write zero bytes for testing...

Reply:
This runs counter to OpenSSL's [documented behavior](https://www.openssl.org/docs/man1.1.1/man3/SSL_write_ex.html), so if the goal is behavioral compatibility I don't think we should diverge unless we need to.

> You should not call SSL_write() with num=0, it will return an error. SSL_write_ex() can be called with num=0, but will not send application data to the peer.

Reply:
Sounds good!